File: Crivodeesrastotenes.c

#include <stdio.h>
#include <stdbool.h>

void crivo(int n) {
    bool ehPrimo[n + 1];
    for (int i = 0; i <= n; i++) {
        ehPrimo[i] = true; // Inicialmente, todos são considerados primos
    }
    ehPrimo[0] = ehPrimo[1] = false; // 0 e 1 não são primos

    for (int p = 2; p * p <= n; p++) {
        if (ehPrimo[p]) {
            for (int i = p * p; i <= n; i += p) {
                ehPrimo[i] = false; // Marcando múltiplos de p como não primos
            }
        }
    }

    // Exibindo números primos
    for (int i = 2; i <= n; i++) {
        if (ehPrimo[i]) {
            printf("%d ", i);
        }
    }
    printf("\n");
}

int main() {
    int n;
    printf("Digite o valor de n: ");
    scanf("%d", &n); // Entrada do valor n
    printf("Números primos até %d:\n", n);
    crivo(n);
    return 0;
}

File: crivo de esrastotenes.py
def crivo(n):
    eh_primo = [True] * (n + 1)
    eh_primo[0] = eh_primo[1] = False  # 0 e 1 não são primos

    for p in range(2, int(n ** 0.5) + 1):
        if eh_primo[p]:
            for i in range(p * p, n + 1, p):
                eh_primo[i] = False  # Marcando múltiplos de p como não primos

    # Exibindo números primos
    for i in range(2, n + 1):
        if eh_primo[i]:
            print(i, end=" ")

n = int(input("Digite o valor de n: "))  # Entrada do valor n
print(f"Números primos até {n}:")
crivo(n)
